sdk/python/kubeflow/training/utils/utils.py

```diff
@@ -117,13 +117,40 @@ def get_script_for_python_packages(
     return script_for_python_packages
 
 
+def generate_container_spec(
+    name: str,
+    image: str,
+    args: Optional[List[str]] = None,
+    resources: Optional[models.V1ResourceRequirements] = None,
+    volume_mounts: Optional[models.V1VolumeMount] = None,
+) -> models.V1Container:
+    """
+    get container spec for given name and image.
+    """
+    if name is None or image is None:
+        container_spec = models.V1Container(name=name, image=image)
+    if args:
+        container_spec.args = args
+
+    if resources:
+        container_spec.resources = resources
+
+    if volume_mounts:
+        container_spec.volume_mounts = volume_mounts
```

Comment:
Can parameters, other than `resources` be None ?

Reply:
Yes, they were none in the current implementation.

Reply:
> Yes, they were none in the current implementation.

@deepanker13 What do you mean by current implementation ? We didn't use this func before.

Reply:
I mean this function is being called in create_job as well, so these parameters are not needed. But when I will call this function in the train api then these will be needed.

Reply:
@deepanker13 But as you can see right now, we don't use this function for existing `create_job` flow: https://github.com/kubeflow/training-operator/blob/765bc00be13ff993203a91dbf67aebed364180b0/sdk/python/kubeflow/training/utils/utils.py#L171-L174
Should we modify it to re-use the `get_container_spec` function  then ?

Reply:
got your point, made the change